stdlib/public/Darwin/Foundation/Data.swift

```diff
@@ -2056,10 +2064,14 @@ public struct Data : ReferenceConvertible, Equatable, Hashable, RandomAccessColl
                 // Copy the contents of buffer...
                 _representation = _Representation(UnsafeRawBufferPointer(start: base, count: endIndex))
 
-                // ... and append the rest byte-wise.
+                // ... and append the rest byte-wise, buffering through an InlineData.
+                var buffer = InlineData()
                 while let element = iter.next() {
-                    Swift.withUnsafeBytes(of: element) {
-                        _representation.append(contentsOf: $0)
+                    if buffer.count < buffer.capacity {
+                        buffer.append(byte: element)
+                    } else {
+                        buffer.withUnsafeBytes {_representation.append(contentsOf: $0)}
+                        buffer.count = 0
                     }
```

Comment:
You don't seem to be draining the buffer at the end... Are we missing unit tests / validation tests here?

Reply:
This is an `InlineData`, meaning that it does not grow or shrink — it's simply a 7- or 15-byte buffer on the stack, with a `UInt8` length; `buffer.count = 0` takes care of "clearing" it out in the sense that we ignore what's already on the stack and overwrite it.

Reply:
AFAICT you are not appending the `buffer` to the `_representation` after the `while` loop exits, dropping its content on the floor.

Reply:
That implies to me, we are missing proper test coverage for this code branch.

Reply:
Good catch — let me fix this and add unit tests to validate.